examples/keyvalue-inmemory/main.go

```diff
@@ -0,0 +1,74 @@
+//go:generate wit-bindgen-wrpc go --out-dir bindings --package github.com/wasmCloud/provider-sdk-go/examples/keyvalue-inmemory/bindings wit
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/wasmCloud/provider-sdk-go"
+	server "github.com/wasmCloud/provider-sdk-go/examples/keyvalue-inmemory/bindings"
+	"golang.org/x/sync/errgroup"
+)
+
+func main() {
+	providerGroup := new(errgroup.Group)
+	p, err := provider.New(
+		provider.SourceLinkPut(handleNewSourceLink),
+		provider.TargetLinkPut(handleNewTargetLink),
+		provider.SourceLinkDel(handleDelSourceLink),
+		provider.TargetLinkDel(handleDelTargetLink),
+		provider.HealthCheck(handleHealthCheck),
+		provider.Shutdown(handleShutdown),
+	)
+	if err != nil {
+		panic(err)
```

Comment:
nit; maybe replace this by a `log.Fatal`?
how about we put this logic into a `run() error`, such that `main` just calls `run` and handles the error gracefully? see https://github.com/wrpc/wrpc/blob/a779107b744f9f986739fd0046c08b546f167164/examples/go/keyvalue-server/cmd/keyvalue-mem-nats/main.go#L173-L177